.github/workflows/test.yml

```diff
@@ -0,0 +1,157 @@
+name: Tests
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - stable/*
+      - release/*
+      - trying
+      - staging
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  integration-tests:
+    name: Integration tests
+    runs-on: "n1-standard-32-netssd-preempt"
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+          cache: 'maven'
+      - uses: stCarolas/setup-maven@v4.3
+        with:
+          maven-version: 3.8.5
+      - run: mvn -T1C -B -DskipChecks -DskipTests package
+      - run: docker build --build-arg DISTBALL=dist/target/camunda-zeebe-*.tar.gz --build-arg APP_ENV=dev -t camunda/zeebe:current-test .
```

Comment:
❌  I am not 100% sure of this one, but if it is an issue we should fix it.

Github's runners guarantee a clean instance each time they are used. With self-hosted runners we don't have this guarantee. I believe this would mean that if we run 2 integration tests jobs at the same time they would both create an image `camunda/zeebe:current-test`. The latest will override the first which could result in strange behaviour since we would be testing against a different image that we'd expect. So unless this runner somehow isolates these jobs we might run into problems.

We could solve this be changing the tag to something else. In ZPT I've been using `${{ github.sha }}` for this.

Reply:
I've found some hints that our self-hosted runners are behaving just like the github-hosted ones: https://github.com/actions-runner-controller/actions-runner-controller#persistent-runners
https://github.com/camunda/infra-core/blob/ac570557c1cdb65d158156c5291ce919418a37cf/camunda-ci/kustomize/base/gha-runner/controller/runners.yml#L35

Reply:
Yes, I use the `ephemeral` runners feature to give them a clean slate and over "leaking" data in the future between different workflows or different repositories even :+1: